akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -484,17 +495,32 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
         // FIXME we should call FragmentAssembler.freeSessionBuffer when image is unavailable
       }
     })
+
     ctx.errorHandler(new ErrorHandler {
+      private val fatalErrorOccured = new AtomicBoolean
+
       override def onError(cause: Throwable): Unit = {
         cause match {
-          case e: ConductorServiceTimeoutException ⇒
-            // Timeout between service calls
-            log.error(cause, s"Aeron ServiceTimeoutException, ${cause.getMessage}")
-
+          case e: ConductorServiceTimeoutException ⇒ handleFatalError(e)
+          case e: DriverTimeoutException           ⇒ handleFatalError(e)
+          case _: AeronTerminated                  ⇒ // already handled, via handleFatalError
           case _ ⇒
             log.error(cause, s"Aeron error, ${cause.getMessage}")
         }
       }
+
+      private def handleFatalError(cause: Throwable): Unit = {
+        if (fatalErrorOccured.compareAndSet(false, true)) {
+          if (!isShutdown) {
+            log.error(cause, "Aeron {}, terminating system. {}", Logging.simpleName(cause), cause.getMessage)
+            taskRunner.stop()
+            aeronErrorLogTask.cancel()
+            system.terminate()
+            throw new AeronTerminated(cause)
+          }
+        } else
+          throw new AeronTerminated(cause)
+      }
```

Comment:
Exactly, that would be rather complicated to try to restart such things. If embedded media driver I can only imagine this happening in case of a very long GC pause (our default for this liveness is 15 sec).

For external media driver it can happen if that process dies, and then the ActorSystem should also go down, in my opinion.